frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/ResetPasswordRequestHandlerTest.java

```diff
@@ -202,6 +202,39 @@ void shouldReturn200AndPutMessageOnQueueForAValidCodeFlowRequest() throws Json.J
                         persistentId);
     }
 
+    @Test
+    void shouldUseExistingOtpCodeIfOneExists() throws Json.JsonException {
+        String persistentId = "some-persistent-id-value";
+        Map<String, String> headers = new HashMap<>();
+        headers.put(PersistentIdHelper.PERSISTENT_ID_HEADER_NAME, persistentId);
+        headers.put("Session-Id", session.getSessionId());
+        Subject subject = new Subject("subject_1");
+        when(authenticationService.getSubjectFromEmail(TEST_EMAIL_ADDRESS)).thenReturn(subject);
+        when(codeStorageService.getOtpCode(any(String.class), any(NotificationType.class)))
+                .thenReturn(Optional.of(TEST_SIX_DIGIT_CODE));
+
+        NotifyRequest notifyRequest =
+                new NotifyRequest(
+                        TEST_EMAIL_ADDRESS, RESET_PASSWORD_WITH_CODE, TEST_SIX_DIGIT_CODE);
+        String serialisedRequest = objectMapper.writeValueAsString(notifyRequest);
```

Comment:
I would move this to below 226 and then add this below to check the same code is added to the queue `verify(awsSqsClient).send(serialisedRequest);`